Modular Exponentiation/modularExponentiation.cpp

```diff
@@ -0,0 +1,35 @@
+#include <bits/stdc++.h>
+using namespace std;
+#define ld long double
+#define ulint unsigned long long int
+#define lint long long int
+#define mp make_pair
+#define pb push_back
+#define f(x) for(lint i=0;i<x;i++)
+#define fj(x) for(lint j=0;j<x;j++)
+#define F first
+#define S second
+#define faster ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0)
+
+lint modex(lint base,lint exponent,lint mod){
+    if(base == 0){
+        return 0;		
+    }
+    if(exponent == 0){
+        return 1;
+    }
+    if(exponent%2 == 0){
+        lint ans = modex(base,exponent/2,mod);
+        return (ans*ans)%mod;
+    }
+    return ((base%mod)*(modex(base,exponent-1,mod)))%mod;
```

Comment:
There's no need to write two different calls for odd and even conditionally. It can be factorised.
Call the function modex(base, exponent/2, mod) and store it (if it is odd, it will anyways send the floor value of exponent). Square the stored value and now if current exponent is odd, just multiply the base with it.
So, a common function call for both.

Reply:
It would anyway have to be checked for odd or even once, which is the case here. So, space and time complexity would remain same.

Reply:
Yes, no issue in that. I just wanted to convey that the code can be refactored here and made a bit more clean (an important aspect of Open Source).

Reply:
I understand. We're on the same page. But also, I do feel that user-readability would be more in this way, as it would allow them to understand the difference in what we are doing for even and odd. What say?

Reply:
Readability, yes that's important. Okay, the implementation is fine then. However, it would be helpful if some comments can be added in the code for this explaining what's actually happening here (if someone is reading modular exponentiation for the 1st time).

Reply:
Yes, done. Please review @gupta2140